torch/distributed/distributed_c10d.py

```diff
@@ -385,6 +394,10 @@ def _new_process_group_helper(world_size,
         raise RuntimeError("The specified group name has already been "
                            "created, please use a different group name")
 
+    if not isinstance(timeout, timedelta):
```

Comment:
Let's move this logic to the very top since it should be the first thing to be checked.